task_manager/task_manager/tasks/mission.py

```diff
@@ -81,15 +81,13 @@ def execute_cb(self, goal_handle: ServerGoalHandle) -> MissionAction.Result:
             mission_result.skipped = False
 
             if subtask_result.task_status != TaskStatus.DONE:
-                if subtask.allow_skipping:
+                if subtask.allow_skipping and not goal_handle.is_cancel_requested:
                     mission_result.skipped = True
                     continue
 
-                if subtask_result.task_status == TaskStatus.CANCELED and goal_handle.is_cancel_requested:
-                    # Need to also check if the cancel was requested. If the goal was cancelled
-                    # through a system task, we cannot set the status to be cancelled and must abort instead.
+                if goal_handle.is_cancel_requested:
                     goal_handle.canceled()
```

Comment:
I'm not sure about this one. If a subtask is still IN_PROGRESS because of the task cancellation failed, or ERROR due to some unexpected problem during the cancellation, we would still mark Mission as "CANCELED". I think the Mission end status should reflect the status of the last executed task. So in these cases it should be ERROR. Having the `subtask_result.task_status == TaskStatus.CANCELED` did this.

Reply:
Yeah that's a good point...didn't consider the cases where the subtask canceling fails and it's forever IN_PROGRESS. Reverted this and that also fixed one of the failing tests